lib/koala/api/graph_error_checker.rb

```diff
@@ -32,18 +35,24 @@ def error_class
           # See: https://developers.facebook.com/docs/authentication/access-token-expiration/
           #      https://developers.facebook.com/bugs/319643234746794?browse=search_4fa075c0bd9117b20604672
           AuthenticationError
+        elsif permission_error?
+          PermissionError
         else
           ClientError
         end
       end
 
       def auth_error?
-        # tbh, I'm not sure why we restrict Facebook-reported OAuthExceptions to only those without
-        # codes or whose codes match the list above -- let's investigate changing this later.
         error_info['type'] == 'OAuthException' &&
           (!error_info['code'] || AUTHENTICATION_ERROR_CODES.include?(error_info['code'].to_i))
       end
 
+      def permission_error?
+        error_info['type'] == 'OAuthException' &&
+          error_info['error_subcode'].nil? &&
```

Comment:
```suggestion
```

I'd not add the check on error_subcode, I'm pretty sure in some cases it is not nil

Reply:
I checked [the Facebook's API documentation](https://developers.facebook.com/docs/marketing-api/error-reference/), seems like some 100 errors with subcode are invalid parameter error. Would we be catching too much errors that some may not necessarily fall under `PermissionError`?

Reply:
Ah indeed the marketing API doesn't have the same error code as the graph api (and codes are probably also differents for instagram api, messenger api and whatsapp api)

I don't think we have a way to differentiate which api is being called to adapt error code handling from within the gem. Maybe it is better not to change anything and let end user handle error themselves

Reply:
Got it, let me close this PR then.